### PR TITLE
199 gallery image count

### PR DESCRIPTION
**Pull request by weiwang314**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,6 +3,12 @@
   "description": "Php library that provides implementations for triniti:curator schemas.",
```
Comment (gdbrown) [inline composer.json]: @weiwang314 this is not needed, remove

File: src/NcrGalleryProjector.php [hunk 2]
```diff
@@ -34,6 +42,108 @@ public static function getSubscribedEvents()
...
+
+        try {
+            /** @var NodeRef $galleryRef */
+            $galleryRef = $node->get('gallery_ref');
+            $imageCount = $this->getImageCount($galleryRef, $pbjx);
+
+            /** @var Gallery $gallery */
+            $gallery = $this->ncr->getNode($galleryRef);
+            $gallery->set('image_count', $imageCount);
+
```
Comment (gdbrown) [inline src/NcrGalleryProjector.php]: @weiwang314 use update and index node method

File: src/NcrGalleryProjector.php [hunk 3]
```diff
@@ -34,6 +42,108 @@ public static function getSubscribedEvents()
...
+            return;
+        }
+
+        try {
+            /** @var NodeRef $galleryRef */
+            $galleryRef = $node->get('gallery_ref');
+            $imageCount = $this->getImageCount($galleryRef, $pbjx);
+
+            /** @var Gallery $gallery */
+            $gallery = $this->ncr->getNode($galleryRef);
```
Comment (gdbrown) [inline src/NcrGalleryProjector.php]: @weiwang314 use same strategy of consistent read, pass context, etc.  ref other methods

File: src/NcrGalleryProjector.php [hunk 4]
```diff
@@ -34,6 +42,108 @@ public static function getSubscribedEvents()
         ];
     }
 
+    /**
+     * @param NodeCreated $event
+     * @param Pbjx $pbjx
+     */
+    public function onAssetCreated(NodeCreated $event, Pbjx $pbjx): void
+    {
+        /** @var Node $node */
```
Comment (gdbrown) [inline src/NcrGalleryProjector.php]: @weiwang314 put guard at start of method to ignore event replays

File: src/NcrGalleryProjector.php [hunk 5]
```diff
@@ -34,6 +42,108 @@ public static function getSubscribedEvents()
...
+        try {
+            /** @var NodeRef $galleryRef */
+            $galleryRef = $node->get('gallery_ref');
+            $imageCount = $this->getImageCount($galleryRef, $pbjx);
+
+            /** @var Gallery $gallery */
+            $gallery = $this->ncr->getNode($galleryRef);
+            $gallery->set('image_count', $imageCount);
+
+            $this->ncr->putNode($gallery);
```
Comment (gdbrown) [inline src/NcrGalleryProjector.php]: @weiwang314 do not wrap in try/catch

File: src/NcrGalleryProjector.php [hunk 6]
```diff
@@ -34,6 +42,108 @@ public static function getSubscribedEvents()
...
+    /**
+     * @param NodeUpdated $event
+     * @param Pbjx $pbjx
+     */
+    public function onAssetUpdated(NodeUpdated $event, Pbjx $pbjx): void
+    {
+        /** @var NodeRef $nodeRef */
+        $nodeRef = $event->get('node_ref');
+
+        $node = $this->ncr->getNode($nodeRef);
```
Comment (gdbrown) [inline src/NcrGalleryProjector.php]: @weiwang314 

- guard to ignore replay
- use the new_node on event itself, don't fetch the noode, it's already there
- remove try catch
- move logic to one method that both create/update can share... `updateImageCount`... which accepts the event, gallery ref and pbjx

File: src/NcrGalleryProjector.php [hunk 7]
```diff
@@ -34,6 +42,108 @@ public static function getSubscribedEvents()
...
+     * @param NodeRef $galleryRef
+     * @param Pbjx $pbjx
+     *
+     * @return int
+     *
+     * @throws \Gdbots\Pbj\Exception\GdbotsPbjException
+     */
+    public function getImageCount(NodeRef $galleryRef, Pbjx $pbjx): int
+    {
+        /** @var SearchAssetsRequest $searchAssetsRequest */
```
Comment (gdbrown) [inline src/NcrGalleryProjector.php]: @weiwang314 fetch only 1 record, in catch return 0., copy context from event to the request

**Comment by weiwang314** [comment]
@gdbrown feedback on review

**Pull request merged**